src/adapters/ui/iconview.js

```diff
@@ -104,17 +105,30 @@ const view = (fileIcon, themeIcon, droppable) => (state, actions) =>
         ]),
         h('div', {
           class: 'osjs-desktop-iconview__entry__label'
-        }, entry.filename)
+        }, (entry.label || entry.filename))
       ])
     ]);
   }));
 
-const createShortcuts = (root, readfile, writefile) => {
+const getApplicationLocaleName = (filename, core) => {
+  const [metadata] = core.make('osjs/packages').getPackages(pkg => (pkg.name === filename));
+  return core.make('osjs/locale').translatableFlat(metadata.title);
+};
+
+const updateShortcutLabel = (shortcut, core) => {
```

Comment:
Instead of mutating the object, create a copy:

```
const updateShortcutLabel = ({label, ...shortcut}, core) => ({
  ...shortcut,
 label: label || (shortcut.mime === 'osjs/application' ? getApplicationLocaleName(shortcut.filename, core) : null)
});
```

Reply:
Saw that you requested a review @bthaase . Mind taking this one as well ?